File: item-list/src/main/java/com/gurunars/item_list/ItemViewBinderEmpty.java
package com.gurunars.item_list;

import android.content.Context;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

class ItemViewBinderEmpty implements EmptyViewBinder {

    static final int EMPTY_TYPE = -404;

    @Override
    public View getView(Context context) {
        TextView view = new TextView(context);
        view.setLayoutParams(new ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT));
        view.setText(R.string.empty);
        view.setGravity(Gravity.CENTER);
        return view;
    }

}

File: item-list/src/main/kotlin/com/gurunars/item_list/ItemViewBinderEmpty.kt
package com.gurunars.item_list

import android.content.Context
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.gurunars.shortcuts.fullSize
import org.jetbrains.anko.matchParent

internal class ItemViewBinderEmpty : EmptyViewBinder {

    override fun getView(context: Context): View {
        return TextView(context).apply {
            fullSize()
            setText(R.string.empty)
            gravity = Gravity.CENTER
        }
    }

    companion object {

        val EMPTY_TYPE = -404
    }

}
